lib/routes/syosetu/index.ts

```diff
@@ -0,0 +1,79 @@
+import { Route, Data, DataItem } from '@/types';
+import { fetchNovelInfo, fetchChapterContent } from './utils';
+import { Context } from 'hono';
+import { NovelType } from 'narou';
+import querystring from 'querystring';
+
+export const route: Route = {
+    path: '/:ncode/:routeParams?',
+    categories: ['reading'],
+    example: '/syosetu/n9292ii',
+    parameters: {
+        ncode: 'Novel code, can be found in URL',
+        routeParams: 'Optional: limit=N (max: 20, default: 5)',
+    },
+    features: {
+        requireConfig: false,
+        requirePuppeteer: false,
+        antiCrawler: false,
+        supportBT: false,
+        supportPodcast: false,
+        supportScihub: false,
+    },
+    name: 'Novel Updates',
+    maintainers: ['eternasuno', 'SnowAgar25'],
+    handler,
+    radar: [
+        {
+            title: 'Novel',
+            source: ['novel18.syosetu.com/:ncode', 'ncode.syosetu.com/:ncode'],
+            target: '/:ncode',
+        },
+    ],
+};
+
+async function handler(ctx: Context): Promise<Data> {
+    const { ncode } = ctx.req.param();
+    const routeParams = querystring.parse(ctx.req.param('routeParams'));
+    const limit = Math.min(Number(routeParams.limit || 5), 20);
```

Comment:
Please make use of the [common parameter limit](https://docs.rsshub.app/guide/parameters#limit-entries) `ctx.req.query('limit')` instead.